cg/apps/demultiplex/sample_sheet/read_sample_sheet.py

```diff
@@ -9,44 +9,37 @@
     SampleSheetNovaSeq6000Sections,
     SampleSheetNovaSeqXSections,
 )
+
 from cg.exc import SampleSheetError
 from cg.io.controller import ReadFile
+import re
 
 LOG = logging.getLogger(__name__)
 
 
-def validate_samples_are_unique(samples: List[FlowCellSample]) -> None:
-    """Validate that each sample only exists once."""
-    sample_ids: set = set()
-    for sample in samples:
-        sample_id: str = sample.sample_id.split("_")[0]
-        if sample_id in sample_ids:
-            message: str = f"Sample {sample.sample_id} exists multiple times in sample sheet"
-            LOG.warning(message)
-            raise SampleSheetError(message)
-        sample_ids.add(sample_id)
-
-
-def get_samples_by_lane(
-    samples: List[FlowCellSample],
-) -> Dict[int, List[FlowCellSample]]:
-    """Group and return samples by lane."""
-    LOG.debug("Order samples by lane")
-    sample_by_lane: Dict[int, List[FlowCellSample]] = {}
-    for sample in samples:
-        if sample.lane not in sample_by_lane:
-            sample_by_lane[sample.lane] = []
-        sample_by_lane[sample.lane].append(sample)
-    return sample_by_lane
-
+def get_sample_sheet_from_file(
```

Comment:
I am wondering if we could apply the single responsibility principle to this function and make it only return the content from the parser and name it `get_sample_sheet_content_from_file`